### ISSUE TITLE
FEAT: Support ipython's `_repr_*_()` protocol

### ISSUE THREAD

**Issue by NickCrews** [microsoft/debugpy#1563]
I often work with objects that support [ipython's `_repr_*_()` protocol](https://ipython.readthedocs.io/en/stable/config/integrating.html#rich-display), such as [rich](https://github.com/Textualize/rich) objects. For example, [Ibis](https://github.com/ibis-project/ibis) returns rich tables as an output format.

If you run this in an ipython terminal, you get nice output:

```python
from rich.__main__ import make_test_card
make_test_card()
```

but in the vscode debugger, you just get `<rich.table.Table object at 0x13cb9b550>`\

Can we use ipython instead of the vanilla interpreter? Or something else to make debugging more ergonomic?

**Comment by int19h** [comment]
This is an interesting idea, but I don't think this is possible currently, since ultimately debugpy just talks DAP which does not accommodate such things directly. The REPL is implemented by the DAP client sending "evaluate" requests.

So, this would need 1) some additions to the DAP spec to allow for rich content transfer and queries, and 2) VSCode actually supporting them on their end.

Technically it would be possible to do this without protocol support utilizing existing "evaluate" requests if the client just sends expressions like `base64.encode(foo.__repr_png__())` etc and then interprets the response data accordingly. It would be Python-specific, however, and VSCode REPL is meant to be generic.

**Comment by NickCrews** [comment]
thanks for the quick response!

I think I can actually greatly reduce the scope of my feature request. How about instead of supporting ALL of the mimetypes like png, we only support `text/plain`? If I understand correctly, this is just plain text that 1) DAP should support and 2) vscode could fairly easily support ([they don't currently, but I want them to](https://github.com/microsoft/vscode-python-debugger/issues/312)).

I'm imagining this could be an implementation that debugpy uses?

```python
from rich.text import Text


def get_text_plain(obj) -> str | None:
    """Get the text/plain representation of an object, if it has one."""
    try:
        bundles = obj._repr_mimebundle_(include=["text/plain"], exclude=[])
    except AttributeError:
        return None
    return bundles.get("text/plain")


def get_repr(obj) -> str:
    if text := get_text_plain(obj):
        return text
    return repr(obj)


obj = Text("Hello\nWorld!", style="bold red")
r = get_repr(obj)  # '\x1b[1;31mHello\x1b[0m\n\x1b[1;31mWorld!\x1b[0m\n'
print(r)  # prints to console in bold red
```

**Comment by int19h** [comment]
Yeah, that would be pretty straightforward. As it happens I am currently working on object repr in debugpy v2, so I'll just integrate this there.

Coincidentally, do you know if IPython - or anyone else - has a repr protocol that incorporates lazy streaming (rather than computing the whole thing at once)?

**Comment by NickCrews** [comment]
@int19h , I'm still interested in this. If you point me to the right places, I can try making a PR
